src/libs/actions/App.ts

```diff
@@ -189,23 +187,23 @@ function getOnyxDataForOpenOrReconnect(isOpenApp = false) {
     }
     return {
         optimisticData: [
-            ...defaultData.optimisticData,
+            ...(defaultData?.optimisticData ?? []),
```

Comment:
I might be missing something obvious, but unsure why we're using optional chaining here (and not below). Can you clarify?

Reply:
@VickyStash Do we also need the nullish coalescing operator? Seems like these will always be defined here.

Reply:
@jjcoffee Since `defaultData` has `OnyxData` type where `optimisticData` is an optional param it gives TS error without nullish coalescing operator:
![image](https://github.com/Expensify/App/assets/23176449/7b94192c-073d-4095-a089-a6268e9ef8b0)

But I think I can try to change the type to` Required<OnyxData>`, it will make more sense in this case and we can get rid of nullish coalescing.

Reply:
Perfect, thanks! I think it's a bit more readable like this now.